### ISSUE TITLE
Add AES-CCM ciphers to all settings

### ISSUE THREAD

**Issue by rhymeswithmogul** [mozilla/server-side-tls#279]
The AES-CCM ciphers and ciphersuites are missing from the Mozilla Server Side TLS document and the SSL Configurator.  This cipher mode was ratified by the IETF in 2012 in [RFC 6655](https://tools.ietf.org/html/rfc6655), and are included in recent versions of OpenSSL.  According to [Wikipedia](https://en.wikipedia.org/wiki/CCM_mode), they may provide better performance on embedded and low-power devices, and the cipher mode is also used in WPA2 CCMP and Bluetooth Low Energy.  Despite poor support by web browsers, they are considered safe, and I feel that they should be added to the list of acceptable ciphers.

TLS 1.3 offers these ciphersuites (available in OpenSSL 1.1.1, but disabled by default):
- `TLS_AES_128_CCM_SHA256`
- `TLS_AES_128_CCM_8_SHA256`

TLS 1.2 offers these ciphers:
- `TLS_ECDHE_ECDSA_WITH_AES_128_CCM`
- `TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8`
- `TLS_ECDHE_ECDSA_WITH_AES_256_CCM`
- `TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8`
- `TLS_DHE_RSA_WITH_AES_128_CCM_8`
- `TLS_DHE_RSA_WITH_AES_256_CCM_8`
- `TLS_DHE_RSA_WITH_AES_128_CCM`
- `TLS_DHE_RSA_WITH_AES_256_CCM`

**Comment by tomato42** [comment]
CCM_8 has weak integrity guarantees so we shouldn't enable them by default, those are useful only for specific environments

adding CCM with the full 16 byte tag it probably a good idea

**Comment by makhomed** [comment]
Related comments from @april about TLS_AES_128_CCM_SHA256 and TLS_AES_128_CCM_8_SHA256 ciphers:

https://github.com/mozilla/ssl-config-generator/issues/124#issuecomment-770002153

> It's not generally enabled by default for most systems, and there is not much reason to enable it by default. People whose clients are embedded systems with low-power ICs that lack crypto acceleration will know to enable it on both ends. This is a small enough group of people to not manually discuss it in a document for general purpose servers.

https://github.com/mozilla/ssl-config-generator/issues/124#issuecomment-770007293

> We also don't list CCM because we don't want people going out of their way to enable them when they're rarely needed. It's often not easy to do so, and by listing them it makes it seem mandatory.

So, recommending to enable TLS_AES_128_CCM_SHA256 by default in the Mozilla Server Side TLS document and the SSL Configurator is not a good idea.

**Comment by gstrauss** [comment]
It may be useful to add a note into Rationale section in https://wiki.mozilla.org/Security/Server_Side_TLS explaining why these ciphers are not included in the guidelines, and some of the explanations above about when an admin might consider adding them.

**Comment by Neustradamus** [comment]
To follow this ticket.

**Comment by makhomed** [comment]
@gstrauss only github repo updated via [#308](https://github.com/mozilla/server-side-tls/pull/308): 

https://github.com/mozilla/server-side-tls/blob/gh-pages/Server_Side_TLS.mediawiki

web site https://wiki.mozilla.org/Security/Server_Side_TLS not updated yet ([Security/Server Side TLS: Revision history](https://wiki.mozilla.org/index.php?title=Security/Server_Side_TLS&action=history)).

**Comment by gstrauss** [comment]
> [@gstrauss](https://github.com/gstrauss) only github repo updated via [#308](https://github.com/mozilla/server-side-tls/pull/308):
> 
> https://github.com/mozilla/server-side-tls/blob/gh-pages/Server_Side_TLS.mediawiki
> 
> web site https://wiki.mozilla.org/Security/Server_Side_TLS not updated yet ([Security/Server Side TLS: Revision history](https://wiki.mozilla.org/index.php?title=Security/Server_Side_TLS&action=history)).

Yes, you are correct.  I have asked someone with access to publish to the wiki.

**Comment by gstrauss** [comment]
FYI: wiki was been re-published with this update

**Comment by makhomed** [comment]
> FYI: wiki was been re-published with this update

Yes, this bug was fixed.

But other bugs still exists:

1) wrong release information on the https://github.com/mozilla/server-side-tls github repo:

<img width="220" height="215" alt="Image" src="https://github.com/user-attachments/assets/b112ce9f-cc37-43d5-be0f-36f1e8f382b5" />

2) wrong release information on the https://github.com/mozilla/server-side-tls/releases page:

<img width="1520" height="815" alt="Image" src="https://github.com/user-attachments/assets/eae60dce-1daf-4df9-813f-932c7e8f6d91" />

Compare it with info on the https://github.com/mozilla/server-side-tls/blob/gh-pages/Server_Side_TLS.mediawiki page:

<img width="1383" height="496" alt="Image" src="https://github.com/user-attachments/assets/76a0d08d-9ebc-4140-adb1-05a32903fe8f" />

**Comment by gstrauss** [comment]
@makhomed thank you for your attentiveness.

The guidelines moved to https://github.com/mozilla/ssl-config-generator/ which is why the releases and tags stopped being updated in this repo after v5.6.  (https://github.com/mozilla/ssl-config-generator/tree/master/src/static/guidelines/)

**Comment by makhomed** [comment]
> [@makhomed](https://github.com/makhomed) thank you for your attentiveness.
> 
> The guidelines moved to https://github.com/mozilla/ssl-config-generator/ which is why the releases and tags stopped being updated in this repo after v5.6. (https://github.com/mozilla/ssl-config-generator/tree/master/src/static/guidelines/)

I am talking not about json files, 
located in the https://github.com/mozilla/ssl-config-generator/tree/master/src/static/guidelines directory, 

but about release version of source code of https://wiki.mozilla.org/Security/Server_Side_TLS wiki page, 
which is now Release [version 6.0](https://wiki.mozilla.org/Security/Server_Side_TLS#Version_History) and located in the https://github.com/mozilla/server-side-tls repository.

Expected Version: `Release 6.0`

Actual Version: `Release 5.6`

Or, if releases info not about source code of https://wiki.mozilla.org/Security/Server_Side_TLS web page,
may be it will be better to delete all garbage releases from repo at all, to avoid misunderstanding and confusion?

**Comment by gstrauss** [comment]
> Or, if releases info not about source code of https://wiki.mozilla.org/Security/Server_Side_TLS web page,
> may be it will be better to delete all garbage releases from repo at all, to avoid misunderstanding and confusion?

Something like that may occur in the near future, in the next month or two.
